### PR TITLE
fix: HTJ2K encoding issues

### PR DESCRIPTION
**Pull request by wayfarer3130**
This issues fixes some problems encoding images at HTJ2K

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -85,7 +85,7 @@ jobs:
             rm -rf /var/lib/apt/lists/*
       - run: gcc --version
       - checkout
-      - run: git submodule update --init --recursive --remote
+      - run: git submodule update --init --recursive
```
Comment (wayfarer3130) [inline .circleci/config.yml:88]: We shouldn't be building non-reproducible builds - the remote setting means use the latest build, which we don't know what it is after building.  Would much rather explicitly update the git extern with specific versions.

File: packages/dicom-codec/src/utils/logger.js [hunk 2]
```diff
@@ -26,8 +26,8 @@ const logger = (() => {
     }
   }
 
-  function setVerbose() {
-    verbose = true;
+  function setVerbose(newVerbose = true) {
```
Comment (wayfarer3130) [inline packages/dicom-codec/src/utils/logger.js:29]: Allows turning on/off logs later on, which helps quite a bit for debugging specific items.

Comment (sedghi) [reply packages/dicom-codec/src/utils/logger.js:29]: is it logger for build or that runs on every decoding?

File: packages/openjphjs/src/HTJ2KEncoder.hpp [hunk 3]
```diff
@@ -29,11 +30,16 @@ class HTJ2KEncoder
...
+                   request_tlm_marker_(false),
+                   set_tilepart_divisions_at_resolutions_(false),
+                   set_tilepart_divisions_at_components_(false),
                    lossless_(true),
                    quantizationStep_(-1.0),
                    progressionOrder_(2), // RPCL
                    blockDimensions_(64, 64)
   {
+    // Use for debugging to ensure that correct encoder is being used
+    OJPH_INFO(0x00010001, "Hello HTJ2K Encoder - v1");
```
Comment (sedghi) [inline packages/openjphjs/src/HTJ2KEncoder.hpp]: can we remove this?

**Review by sedghi — COMMENTED** [review]
Not too much familiar with these changes. Maybe we request Jason and/or Chris too

**Comment by jmhmd** [comment]
I think this all looks fine from the openjphjs perspective, but FYI, Chris and I released an update of openjphjs last week which pulls in the latest OpenJPH library for the wasm build, as well as adding some features. I see that you have some of the changes in here (TLM markers) but it doesn't look like the latest. That may be fine if it's working for cornerstone decode purposes. The latest update does have a breaking change - `setIsUsingColorTransform()` is no longer a function on the encoder, it's set on the imageFrame object. But, I don't think we use that option in DICOMImageLoader decode anyways.

File: packages/dicom-codec/src/utils/logger.js (hunk 2 above)
Comment (wayfarer3130) [reply packages/dicom-codec/src/utils/logger.js:29]: It is on every decoding, and I wanted to be able to turn it on/off for test purposes after load.  Previously, it got turned on at load time if it was on at all, and couldn't be turned off.

File: packages/openjphjs/src/HTJ2KEncoder.hpp (hunk 3 above)
Comment (wayfarer3130) [reply packages/openjphjs/src/HTJ2KEncoder.hpp]: Commented out, but it is really useful to have as commented code to test reload conditions - it isn't obvious when you get a reload necessarily.

**Comment by wayfarer3130** [comment]
@jmhmd - I've updated from your most recent version.  I do have a few fixes already in the C++ code to submit back, mostly error codes to allow better debugging support.

**Comment by jmhmd** [comment]
That's great. I'm sure Chris can review if needed, C++ is above my pay grade :)

**Pull request merged**